### PR TITLE
Make hangman more better

### PR DESCRIPTION
**Pull request by Jerry-py**

### CODE REVIEW

File: disgames/mixins/hangman.py [hunk 1]
```diff
@@ -28,11 +29,28 @@ def _show_guesses(self, embed, guesses):
...
-        word = list(random.choice(words))
+        try:
+            with open('./words.txt', 'r') as file:
+                data = file.read().splitlines()
+                data = random.choice(data)                
+                words = data
+        except:
+            words = []
+            target_url = 'https://raw.githubusercontent.com/andrewthederp/Disgames/main/disgames/mixins/words.txt'
+            async with aiohttp.ClientSession() as session:
```
Comment (MarzaElise) [inline disgames/mixins/hangman.py:40]: You could use `bot.http._HTTPClient__session` for making requests :)

File: disgames/mixins/hangman.py [hunk 2]
```diff
@@ -28,11 +29,28 @@ def _show_guesses(self, embed, guesses):
...
+                with open('./words.txt', 'w') as file:
+                    file.write(data)
+            with open('./words.txt', 'r') as file:
+                data = file.read().splitlines()
+                data = random.choice(data)
+                words = data
+                
+        words = str(words).replace('\n', '')
+                
+        word = list(words)
```
Comment (MarzaElise) [inline disgames/mixins/hangman.py]: These could be split into a different function which *could* be named something like `get_words` or something

You could also use the `bot.http._HTTPClient__session` to make requests

**Review by MarzaElise — APPROVED** [review]

**Comment by Jerry-py** [comment]
So what you want me to do is make the code more better so splitting up the code into functions?

File: disgames/mixins/hangman.py (hunk 2 above)
Comment (Jerry-py) [reply disgames/mixins/hangman.py]: Do I have to do this? Or suggestion?

**Comment by MarzaElise** [comment]
yes

File: disgames/mixins/hangman.py (hunk 2 above)
Comment (MarzaElise) [reply disgames/mixins/hangman.py]: a suggestion, we could change it in the future before another release..

Comment (Jerry-py) [reply disgames/mixins/hangman.py]: Alr

**Comment by Jerry-py** [comment]
Don't mind what I did

**Comment by Jerry-py** [comment]
@andrewthederp Why don't you add dependabot?

File: disgames/mixins/hangman.py (hunk 2 above)
Comment (Jerry-py) [reply disgames/mixins/hangman.py]: Resolved in https://github.com/Jerry-py/Disgames/commit/0576337f4c04e4cb6f49d4c724b475f3d9c84fd5

File: disgames/mixins/hangman.py [hunk 3]
```diff
@@ -8,6 +9,35 @@ class Hangman(commands.Cog):
...
+        response = await self._session.get(self._url)
+        return await response.text()
+    
+    async def _get_word(self):
+        try:
+            with open('./words.txt', 'r') as file:
+                data = file.read().splitlines()
+                word = random.choice(data)                
+        except as Exception:
+            words = await self.request()
```
Comment (MarzaElise) [inline disgames/mixins/hangman.py]: im pretty sure this should be `self._request`

**Review by MarzaElise — APPROVED** [review]
LGTM

**Comment by MarzaElise** [comment]
will resolve merge conflicts and merge it before next release

File: disgames/mixins/hangman.py (hunk 3 above)
Comment (Jerry-py) [reply disgames/mixins/hangman.py]: Will fix

**Pull request merged**